lib/rex/post/meterpreter/ui/console/command_dispatcher/mimikatz.rb

```diff
@@ -8,27 +8,41 @@ module Ui
 
 ###
 #
-# Mimikatz extension - grabs credentials from windows memory.
+# Mimikatz extension - grabs credentials from windows memory (older OSes).
 #
 # Benjamin DELPY `gentilkiwi`
 # http://blog.gentilkiwi.com/mimikatz
 #
 # extension converted by Ben Campbell (Meatballs)
+#
 ###
 class Console::CommandDispatcher::Mimikatz
 
   Klass = Console::CommandDispatcher::Mimikatz
 
   include Console::CommandDispatcher
 
+  #
+  # Name for this dispatcher
+  #
+  def name
+    'Mimikatz'
+  end
+
   #
   # Initializes an instance of the priv command interaction.
   #
   def initialize(shell)
     super
-    if client.arch == ARCH_X86 and client.sys.config.sysinfo['Architecture'] == ARCH_X64
+
+    si = client.sys.config.sysinfo
+    if client.arch == ARCH_X86 and si['Architecture'] == ARCH_X64
+      print_warning('Loaded x86 Mimikatz on an x64 architecture.')
       print_line
-      print_warning "Loaded x86 Mimikatz on an x64 architecture."
+    end
+
+    unless si['OS'] =~ /Windows (NT|XP|2000|2003|\.NET)/i
```

Comment:
Maybe `if si['OS'] !~ /Windows (NT|XP|2000|2003|\.NET)/i` will read better, or maybe that's just me.